README.md

```diff
@@ -1,45 +1,601 @@
+
+PayRequest payRequest = PayRequest.newBuilder()
+        .setStoreId(storeId)
+        .setAmount(amount)
+        .setTerminalId(terminalId)
+        .setShowTips(showTips)
+        .setPaymentMethod(PaymentMethod.newBuilder().setPaymentMethodType(paymentMethodType).build())
+        .build();
+
+PayResponse response = paymentClient.sendPayment(payRequest);
+
+// Note: the response will be returned with a PENDING payment status
+// the payment details should be retrieved by the client until the payment either completes (SUCCESS/FAILED) or is CANCELLED (see below).
+````
+
+### Get Terminal Payment Details
+
+The payment details request requires the following parameters:
+- `storeId` - the ID of your assigned store
+- `orderId` - the Order ID returned in the initial payment response, a unique UUID value for each payment.
+
+#### PaymentDetailsRequest: Payment Details Request
+````java
+public class PaymentDetailsRequest {
+    private String storeId;
+    private String orderId;
+}
+````
+- PayResponse : [Get Payment Detail Response](#payresponse--payment-response)
+
+#### Java Demo
+````java
+import common.PaymentClient;
+import com.kodypay.grpc.pay.v1.*;
+
+PaymentClient paymentClient = new PaymentClient(HOSTNAME, APIKEY);
+
+String storeId = "UUID of assigned store";
+String orderId = "UUID of order generated by Kody";
+
+PaymentDetailsRequest paymentDetailsRequest = PaymentDetailsRequest.newBuilder()
+        .setStoreId(storeId)
+        .setOrderId(orderId)
+        .build();
+
+PayResponse response = paymentClient.getDetails(paymentDetailsRequest);
+````
+
+### Cancel Terminal Payment
+
+#### CancelRequest - Cancel Payment Request
+````java
+public class CancelRequest {
+    private String amount = null;
+    private String orderId = null;
+}
+````
+
+The cancel payment request requires the following parameters:
+
+- `storeId` - the ID of your assigned store
+- `terminalId` - the serial number of the terminal that is processing the payment request
+- `amount` - the amount sent in the original payment request, used to find the payment request
+- `orderId` - the Order ID returned in the initial payment response, a unique UUID value for each payment
+
+#### CancelResponse : Cancel Payment Response
+````java
+public enum PaymentStatus {
+    PENDING("PENDING"),
+    SUCCESS("SUCCESS"),
+    FAILED("FAILED"),
+    CANCELLED("CANCELLED")
+}
+
+public class CancelResponse {
+    private PaymentStatus status;   //Cancellation status
+}
+````
+
+#### Java Demo
+````java
+import common.PaymentClient;
+import com.kodypay.grpc.pay.v1.*;
+
+PaymentClient paymentClient = new PaymentClient(HOSTNAME, APIKEY);
+
+String storeId = "UUID of assigned store"; //STORE_ID
+String terminalId = "Terminal serial number";
+String amount = "1.00";
+String orderId = "UUID of order generated by Kody";
+
+CancelRequest cancelRequest = CancelRequest.newBuilder()
+        .setStoreId(storeId)
+        .setAmount(amount.toPlainString())
```

Comment:
`toPlainString` is used when `amount` is a `BigDecimal` but here it is given as a `String`